src/wallet/wallet.cpp

```diff
@@ -1839,9 +1843,20 @@ CWallet::ScanResult CWallet::ScanForWalletTransactions(const uint256& start_bloc
             WalletLogPrintf("Still rescanning. At block %d. Progress=%f\n", block_height, progress_current);
         }
 
+        bool skip_block{false};
+        if (fast_rescan_filter) {
+            fast_rescan_filter->UpdateIfNeeded();
+            auto match_result{fast_rescan_filter->MatchesBlock(block_hash)};
+            if (match_result.has_value() && !match_result.value()) {
+                result.last_scanned_block = block_hash;
+                result.last_scanned_height = block_height;
+                skip_block = true;
```

Comment:
1b6b09e399773c4bac56afead60e4d29e7fbb010: it may be worth adding a new log category `scan` that outputs which blocks are _not_ skipped. That way I can look up the block hash for missing transactions and see if the problem was the filter or some other step.

Reply:
Thanks, done. As for now, I didn't introduce a new log category, but print it unconditionally via `WalletLogPrintf`.

Reply:
That's probably a bit too much noise for a default log, and also confusing for anyone not familiar with these filters.

Reply:
Agree, I ended up following the suggestion of adding a new log category `scan` now.